interface/sexbound/defeat.lua

```diff
@@ -0,0 +1,34 @@
+require "/scripts/messageutil.lua"
+
+function init()
+  self.config = config.getParameter("config")
+  self.canvas = widget.bindCanvas("interface")
+  self.promise = PromiseKeeper.new()
+  widget.focus("interface")
+end
+
+function update(dt)
```

Comment:
Don't know how much of a load that would create adding a new request promise every second tick, but considering this is a self-renewing abort check, wouldn't it be smarter to make a chain where on success the next promise is chained, so that there is no overlap even with lag?

Reply:
Honestly no idea what this code does or where it comes into play, it's old defeat stuff that I haven't touched. Looking at it, and this is purely a guess, I think this is the special invisible UI that allows for escape with ESC that only turns on for players. So the impact is limited to defeated players with that config option enabled.

That said I don't see why this wouldn't be better as a promise chain, I could make that change.

Reply:
Ah, this portion of the code specifically just dismisses the pane when the node no longer exists. Instead of using promises I just put in this, which I then tested and seems to work great:

`if not status.statusProperty("sexbound_defeated") then
    pane.dismiss()
  end`
  
  As far as dying twice when killed while in a sexnode, I'm not sure exactly what causes it, but I'll dig into that.